Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -509,10 +509,21 @@ bool FullscreenManager::willEnterFullscreen(Element& element)
 
         ancestor->document().resolveStyle(Document::ResolveStyleType::Rebuild);
 
+        auto* renderer = ancestor->renderer();
+        auto* oldContainingBlock = renderer ? renderer->containingBlock() : nullptr;
+
         // Remove before adding, so we always add at the end of the top layer.
         if (ancestor->isInTopLayer())
             ancestor->removeFromTopLayer();
         ancestor->addToTopLayer();
+
+        // resolveStyle() will trigger layout if there are container queries, we need to invalidate
+        // the renderer again since the containing block could have since changed.
+        auto* newContainingBlock = renderer ? renderer->containingBlock() : nullptr;
```

Comment:
is "renderer" guaranteed to be valid after remove/add (not sure what kind of activities they implicitly trigger like style resolve)?